src/wiktextract/config.py

```diff
@@ -96,32 +114,42 @@ def __init__(
         self.expand_tables = expand_tables
         # Some fields for statistics
         self.num_pages = 0
-        self.language_counts = collections.defaultdict(int)
-        self.pos_counts = collections.defaultdict(int)
-        self.section_counts = collections.defaultdict(int)
+        self.language_counts: dict[str, int] = collections.defaultdict(int)
+        self.pos_counts: dict[str, int] = collections.defaultdict(int)
+        self.section_counts: dict[str, int] = collections.defaultdict(int)
         # Some fields related to errors
         # The word currently being processed.
-        self.word = None
-        self.errors = []
-        self.warnings = []
-        self.debugs = []
-        self.redirects = {}
+        self.word: Optional[str] = None
+        self.errors: list[ErrorMessageData] = []
+        self.warnings: list[ErrorMessageData] = []
+        self.debugs: list[ErrorMessageData] = []
+        self.redirects: SoundFileRedirects = {}
         self.data_folder = files("wiktextract") / "data" / dump_file_lang_code
+        self.POS_SUBTITLES: Optional[dict[str, POSSubtitleData]] = None
+        self.POS_TYPES: Optional[set[str]] = None
+        self.LINKAGE_SUBTITLES: Optional[dict[str, str]] = None
+        self.OTHER_SUBTITLES: Optional[dict[str, Union[str, list[str]]]] = None
+        # set the above three in the function below
         self.init_subtitles()
+        self.ZH_PRON_TAGS: Optional[list[str]] = None
```

Comment:
All these variables are never be `None` actually. If they have to be defined here, use empty list/dictionary might be more appropriate.

Reply:
I'll try using pre-declaration (`self.POS_SUBTITLES: dict[str, POSSubtitleData]\n`) and assertions. Should be fine in __init__, this object is created once.